### ISSUE TITLE
Uploading dsym fails with "unsupported file"

### ISSUE THREAD

**Issue by martinhoeller** [getsentry/sentry-cli#2032]
### Environment

What version of the CLI and what OS are you running?

sentry-cli 2.31.0 on macOS 14.4.1

### Steps to Reproduce

run `sentry-cli debug-files check <path_to_archive>.xcarchive/dSYMs/ --log-level=debug`

### Expected Result

I expect dsyms to be supported

### Actual Result

cli says "error: Unsupported file" both for `check` and `upload`

### Logs

```
  DEBUG   2024-04-18 18:12:33.896379 +02:00 sentry-cli version: 2.31.0, platform: "darwin", architecture: "arm64"
  INFO    2024-04-18 18:12:33.896559 +02:00 sentry-cli was invoked with the following command line: "sentry-cli" "debug-files" "check" "<path_to_archive>.xcarchive/dSYMs/" "--log-level=debug"
error: Unsupported file
  INFO    2024-04-18 18:12:33.896772 +02:00 Skipping update nagger update check
```

**Comment by szokeasaurusrex** [comment]
@martinhoeller I was able to reproduce this issue, and I am investigating its cause.

Has this command ever worked in a previous Sentry CLI version? If so, which version(s) did it work with?

And, have you found a workaround?

**Comment by martinhoeller** [comment]
Thanks for investigating @szokeasaurusrex . I've never used Sentry before, so I don't know. I did not find a workaround.

**Comment by szokeasaurusrex** [comment]
Appreciate the information, @martinhoeller!

Also, thank you for reporting this issue – we will try to get it fixed or find a workaround for you as soon as possible!

**Comment by szokeasaurusrex** [comment]
@martinhoeller Just to clarify, are you experiencing the same error with the `sentry-cli debug-files upload` command? If so, please provide the output of the following command (note: you can just pass the entire `xcarchive`, and we recursively search for debug files):

```zsh
sentry-cli debug-files upload "<path_to_archive>.xcarchive/" --log-level=debug
```

Although I could reproduce the error message you are seeing with `sentry-cli debug-files check`, the `upload` command works for me.

**Comment by szokeasaurusrex** [comment]
@martinhoeller, at least in my case, the issue with the `check` command appears to be that the Sentry CLI does not recursively search the `dSYMs` directory for the individual debug files. Instead, it checks whether the directory itself is a debug file, and since the directory is clearly not a debug file (since it is a directory), the command fails. By passing the path to the actual debug file within the `dSYMs` directory, I was able to get the `check` command to work.

On the other hand, the `upload` command _does_ recursively search the path for individual debug files, and so, in my case, the `upload` command was successful.

However, if you are experiencing the same error with the `upload` command, it is possible that the issue you are experiencing is slightly different from the problem I was able to reproduce.

**Comment by martinhoeller** [comment]
Ok, I figured out the problem. The path to the xcarchive contains lots of spaces etc. and sentry-cli's parameter parsing split it up.

The path param is `/Users/martin/Library/Developer/Xcode/Archives/2024-04-18/InYourFace\ AppStore\ \(macOS\)\ 18.04.24,\ 08.46.xcarchiv`

here is the debug output:

```
  DEBUG   2024-04-19 11:04:33.894186 +02:00 sentry-cli version: 2.31.0, platform: "darwin", architecture: "arm64"
  INFO    2024-04-19 11:04:33.894475 +02:00 sentry-cli was invoked with the following command line: "sentry-cli" "debug-files" "upload" "--auth-token" "<token>" "--include-sources" "--org" "blue-banana-software" "--project" "inyourface" "--log-level=debug" "/Users/martin/Library/Developer/Xcode/Archives/2024-04-18/InYourFace" "AppStore" "(macOS)" "18.04.24," "08.46.xcarchive"
  ```

You can see how the path has been split up into several parts. Once I renamed the file it worked.

**Comment by martinhoeller** [comment]
Resolved by putting the path in quotation marks.

**Comment by szokeasaurusrex** [comment]
I think we should improve the `debug-files check` behavior to recursively search the provided directory, but I am glad to hear the `debug-files upload` command works for you!

@martinhoeller Are you still struggling with anything else here? If you are unblocked, I would suggest we close this issue, and I will open a separate issue for improving the `debug-files check` behavior

**Comment by martinhoeller** [comment]
I'm unblocked now and the issue can be closed. Thanks for the quick support!
